### PR TITLE
Improve S1472 (`call-argument-line`): Template tag should be on the same line as the literal

### LINKED ISSUE #3705: Update S1472 (`call-argument-line`): cover scope of ESLint's `no-unexpected-multiline`
https://eslint.org/docs/latest/rules/no-unexpected-multiline

### PR DESCRIPTION
**Pull request by francoismora**
Fixes #3705

### CODE REVIEW

File: src/linting/eslint/rules/call-argument-line.ts [hunk 1]
```diff
@@ -19,29 +19,32 @@
...
     return {
       CallExpression: (node: estree.Node) => {
         const call = node as estree.CallExpression;
         if (call.callee.type !== 'CallExpression' && call.arguments.length === 1) {
-          const sourceCode = context.getSourceCode();
           const parenthesis = sourceCode.getLastTokenBetween(
             call.callee,
             call.arguments[0],
-            token => token.type === 'Punctuator' && token.value === ')',
+            isClosingParen,
```
Comment (yassin-kammoun-sonarsource) [inline src/linting/eslint/rules/call-argument-line.ts:43]: Could you comment on this PR's comment why this change? We already discussed it, but let's record here the reason :)

File: src/linting/eslint/rules/call-argument-line.ts [hunk 2]
```diff
@@ -53,17 +56,40 @@ export const rule: Rule.RuleModule = {
...
+function isNotClosingParen(token: AST.Token) {
+  return !isClosingParen(token);
+}
+
 function reportIssue(
-  loc: { start: estree.Position; end: estree.Position } | estree.Position,
+  loc: { start: Position; end: Position } | Position,
   line: number,
   context: Rule.RuleContext,
+  messageId = 'moveArguments',
```
Comment (yassin-kammoun-sonarsource) [inline src/linting/eslint/rules/call-argument-line.ts:89]: Just a suggestion, but let's remove this default value and be explicit about the message id when calling this function. Probably just nitpicking, but what do you think?

File: src/linting/eslint/rules/call-argument-line.ts [hunk 3]
```diff
@@ -19,29 +19,32 @@
...
           const parenthesis = sourceCode.getLastTokenBetween(
             call.callee,
             call.arguments[0],
-            token => token.type === 'Punctuator' && token.value === ')',
+            isClosingParen,
           );
           const calleeLastLine = (parenthesis ? parenthesis : sourceCode.getLastToken(call.callee))!
             .loc.end.line;
-          const { start } = sourceCode.getTokenAfter(call.callee)!.loc;
+          const { start } = sourceCode.getTokenAfter(call.callee, isNotClosingParen)!.loc;
```
Comment (yassin-kammoun-sonarsource) [inline src/linting/eslint/rules/call-argument-line.ts:47]: There is a code smell here when using the exclamation mark. Let's resolve the code smell as "won't fix" since we know for a fact that there is a token after the function's callee.

**Review by yassin-kammoun-sonarsource — APPROVED** [review]
LGTM! Maybe we should wait for the weird issue on ruling to be fixed before merging. Your call...

File: src/linting/eslint/rules/call-argument-line.ts (hunk 1 above)
Comment (yassin-kammoun-sonarsource) [reply src/linting/eslint/rules/call-argument-line.ts:43]: You can mention this here...

Comment (francoismora) [reply src/linting/eslint/rules/call-argument-line.ts:43]: To cover more cases, the logic determining the caller and argument lines has to use the same 'closing parent' token filter. The rule searches for closing parens to find the caller line and for anything except a closing paren to find the arguments line.
The rule now defines two new functions, `isClosingParen()` and `isNotClosingParen()` for that purpose.

File: src/linting/eslint/rules/call-argument-line.ts (hunk 2 above)
Comment (francoismora) [reply src/linting/eslint/rules/call-argument-line.ts:89]: The default value's purpose was only to ease the reviewer's job by minimizing the changes. I will remove it.

**Comment by francoismora** [comment]
> LGTM! Maybe we should wait for the weird issue on ruling to be fixed before merging. Your call...

Indeed, it's better to wait for the moment.
